headerfs/js.go

```diff
@@ -0,0 +1,402 @@
+//go:build js && wasm
+
+package headerfs
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall/js"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcwallet/walletdb"
+	"github.com/linden/indexeddb"
+)
+
+const headersStore = "headers"
+
+// headerStore combines a IndexedDB store of headers within a flat file in addition
+// to a database which indexes that flat file. Together, these two abstractions
+// can be used in order to build an indexed header store for any type of
+// "header" as it deals only with raw bytes, and leaves it to a higher layer to
```

Comment:
I think we should revise this `godoc` a bit to note the high level structure of this impl. IIUC, it maps `height => header` within `indexeddb`. It's the based off a similar bucket structure as boltdb, with the top level key here being  `headersStore`.